auth/auth.go

```diff
@@ -224,13 +231,61 @@ func (c *Client) VerifyIDTokenAndCheckRevoked(ctx context.Context, idToken strin
 		return nil, err
 	}
 
-	user, err := c.GetUser(ctx, p.UID)
+	revoked, err := c.checkRevoked(ctx, p)
 	if err != nil {
 		return nil, err
 	}
-
-	if p.IssuedAt*1000 < user.TokensValidAfterMillis {
+	if revoked {
 		return nil, internal.Error(idTokenRevoked, "ID token has been revoked")
 	}
 	return p, nil
 }
+
+// VerifySessionCookie verifies the signature and payload of the provided Firebase session cookie.
+//
+// VerifySessionCookie accepts a signed JWT token string, and verifies that it is current, issued for the
+// correct Firebase project, and signed by the Google Firebase services in the cloud. It returns a Token containing the
+// decoded claims in the input JWT. See https://firebase.google.com/docs/auth/admin/manage-cookies for more details on
+// how to obtain a session cookie.
+//
+// This function does not make any RPC calls most of the time. The only time it makes an RPC call
+// is when Google public keys need to be refreshed. These keys get cached up to 24 hours, and
+// therefore the RPC overhead gets amortized over many invocations of this function.
+//
+// This does not check whether or not the cookie has been revoked. Use `VerifySessionCookieAndCheckRevoked()`
+// when a revocation check is needed.
+func (c *Client) VerifySessionCookie(ctx context.Context, sessionCookie string) (*Token, error) {
+	return c.cookieVerifier.VerifyToken(ctx, sessionCookie)
+}
+
+// VerifySessionCookieAndCheckRevoked verifies the provided session cookie, and additionally checks that the
+// cookie has not been revoked.
+//
+// This function uses `VerifySessionCookie()` internally to verify the cookie JWT. However, unlike
+// `VerifySessionCookie()` this function must make an RPC call to perform the revocation check.
+// Developers are advised to take this additional overhead into consideration when including this
+// function in an authorization flow that gets executed often.
+func (c *Client) VerifySessionCookieAndCheckRevoked(ctx context.Context, sessionCookie string) (*Token, error) {
+	p, err := c.VerifySessionCookie(ctx, sessionCookie)
+	if err != nil {
+		return nil, err
+	}
+
+	revoked, err := c.checkRevoked(ctx, p)
+	if err != nil {
+		return nil, err
+	}
+	if revoked {
+		return nil, internal.Error(sessionCookieRevoked, "session cookie has been revoked")
+	}
+	return p, nil
+}
+
+func (c *Client) checkRevoked(ctx context.Context, token *Token) (bool, error) {
+	user, err := c.GetUser(ctx, token.UID)
+	if err != nil {
+		return false, err
+	}
+
+	return token.IssuedAt*1000 < user.TokensValidAfterMillis, nil
```

Comment:
Does this line need to handle skew? Could use a comment either way.

Reply:
I feel like we should, but our other implementations of this API currently doesn't do it. I will leave it as is for now, but run it by the Auth team for some comments.